src/include/stir/recon_buildblock/ProjMatrixByBin.inl

```diff
@@ -104,11 +119,85 @@ get_proj_matrix_elems_for_one_bin(
 #endif
         cache_proj_matrix_elems_for_one_bin(probabilities);
       }
-      symm_ptr->transform_proj_matrix_elems_for_one_bin(probabilities);
+      if ( proj_data_info_sptr->is_tof_data() &&
+                                   this->tof_enabled)
+      {
+          LORInAxialAndNoArcCorrSinogramCoordinates<float> lor;
+          proj_data_info_sptr->get_LOR(lor, bin);
+          LORAs2Points<float> lor2(lor);
+          probabilities.set_bin(bin);
+          // now apply TOF kernel and transform to original bin
+          apply_tof_kernel_and_symm_transformation(probabilities, lor2.p1(), lor2.p2(), symm_ptr);
+
+      }
+      else
+      {
+          // now transform to original bin
+          symm_ptr->transform_proj_matrix_elems_for_one_bin(probabilities);
+      }
       cache_proj_matrix_elems_for_one_bin(probabilities);      
     }
   }  
   // stop_timers(); TODO, can't do this in a const member
 }
 
+void
+ProjMatrixByBin::apply_tof_kernel_and_symm_transformation(ProjMatrixElemsForOneBin& probabilities,
+                                  const CartesianCoordinate3D<float>& point1,
+                                  const CartesianCoordinate3D<float>& point2,
+                                  const unique_ptr<SymmetryOperation>& symm_ptr)  STIR_MUTABLE_CONST
+{
+
+    CartesianCoordinate3D<float> voxel_center;
+    float new_value = 0.f;
+    float low_dist = 0.f;
+    float high_dist = 0.f;
+
+    // The direction can be from 1 -> 2 depending on the bin sign.
+    const CartesianCoordinate3D<float> middle = (point1 + point2)*0.5f;
+    const CartesianCoordinate3D<float> diff = point2 - middle;
+
+    const float lor_length = 1.f / (std::sqrt(diff.x() * diff.x() +
+                                        diff.y() * diff.y() +
+                                        diff.z() * diff.z()));
+
+    for (ProjMatrixElemsForOneBin::iterator element_ptr = probabilities.begin();
+         element_ptr != probabilities.end(); ++element_ptr)
+    {
+        Coordinate3D<int> c(element_ptr->get_coords());
+        symm_ptr->transform_image_coordinates(c);
+
+        voxel_center =
+                image_info_sptr->get_physical_coordinates_for_indices (c);
+
+        project_point_on_a_line(point1, point2, voxel_center);
```

Comment:
From geometry, I think that this step is not needed. The `inner_product` below will be equal before or after projection (`x1.x2 = norm(x1)norm(x2)cos(theta)`, so we get the `cos(theta)` factor without having to project first).

Reply:
we could try
```
const float d2_alt = -inner_product(image_info_sptr->get_physical_coordinates_for_indices (c) - middle, diff) * lor_length;
```